tests/api/handlers_test.py

```diff
@@ -413,3 +416,38 @@ def test_public_keys(session, users, http_client, base_url):  # noqa: F811
     assert rows[0]["fingerprint"] == "e9:ae:c5:8f:39:9b:3a:9c:6a:b8:33:6b:cb:6f:ba:35"
     assert rows[0]["fingerprint_sha256"] == "MP9uWaujW96EWxbjDtPdPWheoMDu6BZ8FZj0+CBkVWU"
     assert rows[0]["comment"] == "some-comment"
+
+
+@pytest.mark.gen_test
+def test_request_logging(session, users, http_client, base_url):  # noqa: F811
+    """Test that the api request handlers properly log stats"""
+    # It does not appear possible to add a mock plugin to the global plugins here and expect it to
+    # be called by the request handler: the server side will create its own plugin proxy
```

Comment:
It's created in the `api_app` fixture. This code is ugly because it should be converted to an itest, but for now in legacy code you can get at the plugin proxy by injecting the `app` fixture and then adding plugins to `app.plugins`. That will let you get rid of the mocking, which will make this more straightforward.

Reply:
Ugh, this is more complicated than I thought now that I've looked at it more closely, since we do inject a proper separate `PluginProxy` to the API server right now. I think you need to also modify `tests/fixtures.py` in `api_app` to call `set_global_plugin_proxy` for right now until we come up with a better way.